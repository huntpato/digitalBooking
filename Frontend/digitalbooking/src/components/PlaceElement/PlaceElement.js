import React from 'react'
import styles from './PlaceElement.module.css'

export const PlaceElement = props => {

    const {elemento, locationtext, matching} = styles;
    const {city, search} = props;
    const location = city.split(",");
    const match = location[0].match(new RegExp(`(${search})`, 'gi'));
    const textReplace = location[0].replace(new RegExp(`(${search})`, 'gi'), '*');
 
    return (
        <li className = {elemento}>
            <i className="fas fa-map-marker-alt"></i>
            <div className = {locationtext}>
            <h5>{textReplace.split('').map((element=>{
                if(element === '*'){
                    return <strong className={matching}>{match[0]}</strong>
                } else{
                    return element
                } 
            }))}</h5>
            <h6>{location[1]},{location[2]}</h6>
            <hr/>
            </div>    
        </li>
    )
}