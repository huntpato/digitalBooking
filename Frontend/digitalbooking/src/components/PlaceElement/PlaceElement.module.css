.elemento{
    display: flex;
    flex-direction: row;
    align-items:center;
    list-style-type:none ;
    margin: 0;
    padding-top: 12px;
    padding-bottom: 5px;

}

.elemento i{
    color:var(--color-typo2);
}

.locationtext{
    width: 95%;
}

.locationtext h5{
    color: var(--color-typo2);
    font-size: 12px;
}


.locationtext h6{
    color: var(--color-typo2);
    font-size: 10px;
    padding-bottom: 3px;
}

hr{
    background-color:var(--principal-color);
    border: none;
    width: 100%;
    height: 1.2px;
}

.matching{
    color: var(--principal-color);
    font-size: 13.3px;
    font-weight:bolder;
}

@media (max-width: 576px){

    .locationtext h5{

        font-size: 14px;
    }

    .locationtext h6 {

        font-size:12px;
        padding-bottom: 3px;
    }

}

@media (max-width: 480px){

    .locationtext h5{

        font-size: 14px;
    }

    .locationtext h6 {

        font-size:12px;
        padding-bottom: 3px;
    }

    hr{
        width: 100%;
    }

}