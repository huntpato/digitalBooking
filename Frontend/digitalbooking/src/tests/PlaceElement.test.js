
import React from 'react';
import { shallow } from 'enzyme';
import { PlaceElement } from '../components/PlaceElement/PlaceElement'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <PlaceElement />', ()=> {
  
    const city= "Buenos Aires, Buenos Aires, Argentina"
    const wrapper= shallow(<PlaceElement city={city}/>);

    test('Comprobar que <PlaceElement /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })


    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const li = wrapper.find('li');
        expect(li.hasClass('elemento')).toBe(true);
    })



})
