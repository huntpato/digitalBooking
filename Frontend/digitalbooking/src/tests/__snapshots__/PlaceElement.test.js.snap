// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <PlaceElement /> Comprobar que <PlaceElement /> renderiza correctamente. 1`] = `
<li
  className="elemento"
>
  <i
    className="fas fa-map-marker-alt"
  />
  <div
    className="locationtext"
  >
    <h5>
      B
      u
      e
      n
      o
      s
       
      A
      i
      r
      e
      s
    </h5>
    <h6>
       Buenos Aires
      ,
       Argentina
    </h6>
    <hr />
  </div>
</li>
`;
